pulp_container/tests/functional/api/test_pull_content.py

```diff
@@ -165,6 +167,17 @@ def test_api_performes_schema_conversion(self):
         base_content_type = content_response.headers["Content-Type"].split(";")[0]
         self.assertIn(base_content_type, {MEDIA_TYPE.MANIFEST_V1, MEDIA_TYPE.MANIFEST_V1_SIGNED})
 
+        header_digest = content_response.headers["Docker-Content-Digest"]
+        converted_manifest = json.loads(content_response.content)
+        converted_manifest.pop("signatures")
+        manifest_string = json.dumps(
+            converted_manifest, indent=3, sort_keys=True, separators=(",", ": ")
+        ).encode("utf-8")
```

Comment:
Is this the approved way to write the manifest in order to produce the correct signature?

Reply:
Well, I suppose, yes. Since we do the same operation in the reverse order when we are building a converted manifest: https://github.com/pulp/pulp_container/blob/070238111d9e6b297272f94ac1e8d9bc42c89c1b/pulp_container/app/schema_convert.py#L215-L222

https://github.com/pulp/pulp_container/blob/070238111d9e6b297272f94ac1e8d9bc42c89c1b/pulp_container/app/schema_convert.py#L203-L204

Reply:
Than we could use the same code in line 229 of schema_conversion.py instead of the rather hacky string manipulation.

Reply:
No, because the signature is usually appended to the bottom of the payload. In the `dumps` call, there is used the argument `sort_keys=True`. This would change the order of the keys.

The base64 encoded field `formatLength` (encoded in the field `protected`) present the signature describes how many bytes correspond to the payload. Once we change the order, the clients could trim out the contents of the actual payload.

In the test, we can call the same command because we popped the signature from a dictionary-like object. Not all container clients will parse the content in a similar way in the future, right?

Reply:
Ah, i see.